src/haddock/gear/yaml2cfg.py

```diff
@@ -56,7 +57,7 @@ def _yaml2cfg_text(ycfg, module, explevel):
     params = []
     exp_levels = {
         _el: i
-        for i, _el in enumerate(config_expert_levels + ("all",))
+        for i, _el in enumerate(config_expert_levels + ("all", _hidden_level))
```

Comment:
It's interesting to see that the actual addition that prevents parameters to be written to `cfg` files is this single addition.